### ISSUE TITLE
invalid login type

### ISSUE THREAD

**Issue by ronny35** [robrotheram/taiga-contrib-openid-auth#35]
Hi,

I am using Docker image with OpenID auth preinstallation but, after configuration (Keycloak urls), I have this error : invalid login type, even if login process is successful in Keycloak.

Could you help me on this topic ?

**Comment by mphcomputing** [comment]
did you ever figure this out? I'm having the same issue.

**Comment by ronny35** [comment]
No, this issue still occurs

**Comment by dcabecinhas** [comment]
Bump. Same with me with the `robrotheram/taiga-*-openid` images.

**Comment by adrha** [comment]
Have you set `ENABLE_OPENID: "True"`?

**Comment by dcabecinhas** [comment]
@adrha Yes. I have `ENABLE_OPENID: "True"` on the backend and `ENABLE_OPENID: "true"` on the frontend.
I get redirected to my SSO solution, my credentials are validated, but upon returning to Taiga I get "invalid login type" error.

**Comment by adrha** [comment]
Okay, I experienced the same issue with the login-type error, when I forgot to enable the OIDC.

Since then, I'm getting an error that the OIDC information could not be read from the token (after a very long timeout when Keycloak redirected me after login).
This issue is also an open issue from someone else here in the project.

**Comment by shibacomputer** [comment]
This project doesn't work at all for me either, and provides almost zero context as to why.

**Comment by djsime1** [comment]
It took some digging, but I think I figured out why: By default, this plugin expects your IDP's [`sub` OpenID claim](https://openid.net/specs/openid-connect-core-1_0.html#IDToken) to match the Taiga user ID (the number, not the username) exactly. This is almost never the case unless the users are registering via the plugin. If a user with a matching username/email exists, Taiga will not create a new one and instead throw an error. The code to handle this situation is already present in [taiga-contrib-github-auth](https://github.com/kaleidos-ventures/taiga-contrib-github-auth/), which searches for an existing user with a matching email. If this plugin was "heavily based" on that, then why it wasn't ported is beyond me.

I'm going to work on a fork/PR that addresses this issue. In the mean time, if you can set custom user attributes, create one that maps the user's ID (from Taiga's admin panel) to some field name, then change the `OPENID_ID_FIELD` environment variable to that field.

**Comment by sridharmass** [comment]
Do you have any updates on this?
